packages/core/src/Stores/Modules/Cashier/on-ramp-store.js

```diff
@@ -11,72 +11,59 @@ export default class OnRampStore extends BaseStore {
+    @observable is_widget_html_loading = false;
+    @observable onramp_providers = [];
     @observable selected_provider = null;
     @observable should_show_widget = false;
+    @observable widget_error = null;
+    @observable widget_html = null;
 
     deposit_address_ref = null;
 
     constructor(root_store) {
         super({ root_store });
-        this.onramp_providers = [
-            {
-                default_from_currency: 'usd',
-                description: (
-                    <Localize i18n_default_text='Your simple access to crypto. Fast and secure way to exchange and purchase 150+ cryptocurrencies. 24/7 live-chat support.' />
-                ),
-                from_currencies: ['usd', 'eur', 'gbp'],
-                widget_script_dependencies: { 'changelly-affiliate-js': 'https://widget.changelly.com/affiliate.js' },
-                getWidgetHtml() {
-                    const currency = getCurrencyDisplayCode(root_store.client.currency).toLowerCase();
-                    return `<iframe src="https://widget.changelly.com?from=${this.from_currencies.join(
-                        ','
-                    )}&to=${currency}&amount=50&address=&fromDefault=${
-                        this.default_from_currency
-                    }&toDefault=${currency}&theme=danger&merchant_id=iiq3jdt2p44yrfbx&payment_id=&v=2" width="100%" height="475px" class="changelly" scrolling="no" onLoad="function at(t){var e=t.target,i=e.parentNode,n=e.contentWindow,r=function(){return n.postMessage({width:i.offsetWidth},it.url)};window.addEventListener('resize',r),r()};at.apply(this, arguments);" style="min-height: 100%; min-width: 100%; overflow-y: visible; border: none">Can't load widget</iframe>`;
-                },
-                icon: 'IcCashierChangelly',
-                name: 'Changelly',
-                payment_icons: ['IcCashierVisa', 'IcCashierMastercard'],
-                to_currencies: ['bch', 'btc', 'etc', 'eth', 'ltc', 'ust'],
-            },
-        ];
-    }
-
-    @action.bound
-    onMount() {
-        this.disposeThirdPartyJsReaction = reaction(
-            () => this.selected_provider,
-            provider => {
-                if (!provider?.widget_script_dependencies) return;
-
-                Object.keys(provider.widget_script_dependencies).forEach(script_name => {
-                    if (!document.querySelector(`#${script_name}`)) {
-                        const el_script = document.createElement('script');
-                        el_script.src = provider.widget_script_dependencies[script_name];
-                        el_script.id = script_name;
-                        document.body.appendChild(el_script);
-                    }
-                });
-            }
-        );
-    }
 
-    @action.bound
-    onUnmount() {
-        if (typeof this.disposeThirdPartyJsReaction === 'function') {
-            this.disposeThirdPartyJsReaction();
-        }
+        this.onClientInit(async () => {
```

Comment:
doesn't seem to be awaiting for anything so `async` could be removed here?

Reply:
`onClientInit` expects a `Promise` to be returned (see `deriv-app/packages/core/src/Stores/base-store.js`).